### DISCUSSION TITLE
2022-11-08 17:21:10の試合で東4局の後に東1局が行われているように見えます

### DISCUSSION THREAD

**Discussion by oshizo** [smly/mjai.app#16] [General]
楽しいサービスを作っていただきありがとうございます。

牌譜ビューワーでログを見ていたのですが、2022-11-08 17:21:10のゲームの「席順2」の半荘で、東4局3本場の後に東1局が再度行われているように見えます。
その後、再び同じように東4局3本場に進んでいますが、この「2回目の」東4局3本場では、終局時に捨て牌に1sが7枚見えているなど、正常ではない状態に見えました。

また、これに関係するかは分からないのですが、同じゲームの、同じ「席順2」の半荘で試合結果に表示されている点数が合計10万点より3万点近く過少になっています。
試合結果の表示は最終局である南4局2本場でのビューワーの点数表示ともかなり異なっているのですが、これは、ペナルティを受けた参加者がいる時にこのような点数の見え方をするのでしょうか。
「試合結果」の表示 ... 起家:(19400), 下家:(19600), 対面:(16800), 上家:(15200)
「南4局2本場」終局時のビューワーでの表示 ... 起家:(15900), 下家:(30700), 対面:(24500), 上家:(12900)
この席順2の半荘では、東4局2本場の終局時の表示に「ロン」「ツモ」「流局」のいずれも表示されておらず、ここでいずれかのプレイヤーがペナルティをうけたのかと想像しました。もしそうであれば、各局の終了時点のビューワーと、各席順の試合結果の表示欄に、どのプレイヤーがペナルティを受けたのか分かる表示があるとありがたいです。

**Comment by smly** [comment]
ゲームへの参加と詳細の報告をあげていただきありがとうございます！
私事ながら書籍校正と登壇準備の予定が重なり12月まで対応が難しそうです。気長に待っていただけると助かります:bow:

* bot のエラーによる対局中断時に牌譜がおかしくなる問題（あわせてビューワーの表示がおかしくなる問題）
* bot 開発者にとってエラーが発生している試合を把握してデバッグすることが困難

大きく分けて上記2点、今後改善に着手します。
